### PR TITLE
Dev avtka screen

### PR DESCRIPTION
**Pull request by harryhaaren**
Bunch of patches to improve the screen handling APIs. Primary change is that the screen-updates are now callback driven, and Ctlra library manages the requested FPS and timers. This removes that complexity from the applications, and (in theory) allows Ctlra to get smart about how/when to issue the larger USB bulk xfers for screen updates.

Some other changes are in this PR too;
- A `select_device` example, on how to use the get Vendor/Device APIs, and show them in dropdowns (gtk3 based)
- USB backend statistic improvements
- added Valgrind suppresion file for JACK functions
- MIDI generic backend disconnect() midi-backend leak fix
- some maschine mk2 #include fixes

### CODE REVIEW

File: ctlra/devices/avtka.c [hunk 1]
```diff
@@ -212,6 +231,32 @@ avtka_mirror_hw_cb(struct ctlra_dev_t* base, uint32_t num_events,
...
+	/* TODO update Ctlra API to return screen data for ID */
+	uint8_t screen = 0;
+	if(screen >= CTLRA_NUM_SCREENS_MAX)
+		return -2;
+
+	struct avtka_screent_t *scr = &dev->screen[screen];
+	*pixels = avtka_screen_get_data_ptr(a, screen);
+	/* TODO: fix hard coded size */
+	*bytes = ((scr->h_px * scr->w_px) * scr->bits_per_pixel) / 8;
+	//*bytes = (128 * 64 / 8);
```
Comment (foolswood) [inline ctlra/devices/avtka.c:254]: Commented out code.

File: ctlra/usb.c [hunk 2]
```diff
@@ -695,9 +695,66 @@ int ctlra_dev_impl_usb_bulk_write(struct ctlra_dev_t *dev, uint32_t idx,
...
+				       size,
+				       ctlra_usb_xfr_write_done_cb,
+				       dev, /* userdata - pass dev to
+					       banish it if required */
+				       timeout);
+	if(libusb_submit_transfer(xfr) < 0) {
+		libusb_free_transfer(xfr);
+		free(usb_data);
+		dev->usb_xfer_counts[USB_XFER_BULK_ERROR]++;
+		//printf("error submitting data!!\n");
```
Comment (foolswood) [inline ctlra/usb.c:747]: Commented out code.

**Review by foolswood — APPROVED** [review]
Didn't spot anything off in my flick through.

**Comment by harryhaaren** [comment]
This PR is superseeded by the full screen API/callback/partial-redraw PR here: https://github.com/openAVproductions/openAV-Ctlra/pull/74

Closing.

**Pull request closed without merging**